homeassistant/components/media_player/cast.py

```diff
@@ -427,3 +430,16 @@ def async_set_chromecast(self, chromecast):
 
         self.cast_status = self.cast.status
         self.media_status = self.cast.media_controller.status
+
+    @asyncio.coroutine
+    def async_will_remove_from_hass(self):
+        """Disconnect Chromecast object when removed."""
+        self._async_disconnect()
```

Comment:
You probably also should remove the entity object from `hass.data[ADDED_CAST_DEVICE_KEY]` here (or in `_async_disconnect`?)

Reply:
I'm kinda already doing that on line 129, but I think I should move that into `_async_disconnect`, as it makes more sense there.

Reply:
Oh, yes. But that line is never executed if `async_will_remove_from_hass` is being called from "outside", i.e., not because of a re-discovered entity. Putting it here or in `_async_disconnect` is the right thing, I guess.